lib/subscription.js

```diff
@@ -24,7 +24,15 @@ function createConnectionHandler ({ schema, subscriber, fastify, onConnect, lruG
     let context = {}
     if (subscriptionContextFn) {
       try {
-        context = await subscriptionContextFn(connection, request)
+        context = subscriptionContextFn(connection, request)
+
+        if (context instanceof Promise) {
```

Comment:
The async error was not being catched without doing this

Reply:
I don't understand why using `await` would not have worked. can you articulae?

Reply:
awaiting the context here doesn't let the socket connection be properly created, the possible promise should be started, but it's result is not used until the socket messages 'connection_init' / 'start'.

Reply:
Then let's move where we do this operation. If the context should be created and awaited during `connection_init`, then let's do it there. This adds complexity that could bite us in the future.

Reply:
The thing is, as I mentioned, should the context be created at socket creation or in `connection_init`?, as the test `subscription connection is closed if context function **throws**` suggests, it's still possible to make the context without the `connection_init` being received.

As I said previously:

> EDIT: The Context should be made (and it's possible errors, handled) in the request context, and the subcription connection should just receive the promise, since like the test 'subscription connection is closed if context function throws' in `subscription.js` test file suggests, it's possible to start the socket connection first without the 'connection_init' and the context should already be in process. Just passing the function to be executed in `subcription-connection.js` delays the promise unnecessarily.

Reply:
I would move it to https://github.com/mcollina/fastify-gql/pull/234/files#diff-d0a65557fba21a77676fdc6c8ab721c7R95

Reply:
Then I will have to modify the mentioned test and give a function to the subscription connection that resolves the context instead of just receiving it.